templates/plugin/salt_master.conf.erb

```diff
@@ -0,0 +1,56 @@
+# /etc/salt/master.d/foreman.config Master configuration
+#
+# This file summarizes configurations for the salt-master.
+# Have a look at the [Foreman Salt Plugin Documentation](https://theforeman.org/plugins/foreman_salt/) for detailed explanations.
+#
+# After editing this file, run the following command to active the changes:
+# systemctl restart salt-master
+
+
+##
+# Autosign
+autosign_grains_dir: <%= scope.lookupvar('foreman_proxy::plugin::salt::autosign_grains_dir') %>
+autosign_file: <%= scope.lookupvar('foreman_proxy::plugin::salt::autosign_file') %>
+# Uncomment the next line to make use of the autosign host name file (not recommended)
+# permissive_pki_access: True
+
+
+##
+# Node classifier
+master_tops:
+  ext_nodes: /usr/bin/foreman-node
+
+
+##
+# Pillar data access
+ext_pillar:
+  - puppet: /usr/bin/foreman-node
+
+
+##
+# Salt API access
+external_auth:
+  <%= scope.lookupvar('foreman_proxy::plugin::salt::api_auth') %>:
+    <%= scope.lookupvar('foreman_proxy::plugin::salt::api_username') %>:
+      - '@runner'
+
+rest_cherrypy:
+  port: 9191
+  ssl_key: <%= @foreman_ssl_key %>
+  ssl_crt: <%= @foreman_ssl_cert %>
```

Comment:
This config file is written to `/etc/salt/master.d/foreman.conf`. Do you run the Salt master as the `foreman-proxy` user or add the actual user to the `foreman-proxy` group?

Reply:
During my testing, I'm starting the service directly with root.
The [manual](https://theforeman.org/plugins/foreman_salt/13.0/index.html#2.1.1SystemConfiguration) adds a sudoers entry for the `foreman-proxy` user. Since only the user and not the whole group is added here, I guess it is supposed to be executed via the `foreman-proxy` user and _not_ by every user of that group.

Is this important for the file permission or when should we pay attention on it?

Reply:
For security it's important to limit the privileges as much as possible. Running a service as root is considered a bad practice. For production deployments I'd encourage you to run it as a user with limited permissions.

Reply:
Alright, sounds totally reasonable!
Should we change something in this PR  or can we take any action here?

Reply:
Perhaps it's something to address in a follow up PR?